parser/parser.coffee

```diff
@@ -126,6 +127,18 @@ parseClassExpression = ->
   new Class expressions
 
 
+parseTraitExpression = ->
+  match 'trait'
+  match ':'
+  match 'newline'
+  expressions = []
+  while not at 'end'
+    expressions.push parseExpression()
```

Comment:
I don't think nested classes and traits are an issue - they're basically fancy maps anyway.